packages/core/bootstrap/src/lib/cache/index.ts

```diff
@@ -175,7 +177,19 @@ export const withCache: Middleware = async (execute, options = defaultOptions())
         logger.debug(`Cache: GET ${key}`, entry)
         const reqMaxAge = _getRequestMaxAge(data)
         if (reqMaxAge && reqMaxAge !== entry.maxAge) await _cacheOnSuccess(entry)
-        return { jobRunID: data.id, ...entry }
+        const ttl = await cache.ttl(key)
+        const staleness = entry.maxAge - ttl
+        const debug = {
+          cacheHit: true,
+          staleness,
+          performance: endMetrics(true, staleness),
+          providerCost: 0,
+        }
+        return {
+          jobRunID: data.id,
+          ...entry,
+          debug,
```

Comment:
If any other middleware adds debug info to `entry`, they will now be overwritten.

Consider doing something like:

 ```javascript
return {
  jobRunID: data.id,
  ...entry,
  debug: {
    ...(entry.debug || {}),
    ...debug
  }
}

Reply:
Like you did below for result, the same should be done for this flow.

Reply:
`entry` is the cache data, it should never have `debug` in it, no?